lib/utils/request.ts

```diff
@@ -0,0 +1,68 @@
+import { request as nodeHttpRequest } from "http";
+import { request as nodeHttpsRequest } from "https";
+import {
+  supportsSendBeacon,
+  supportsXMLHttpRequest,
+  supportsNodeHttpModule
+} from "./featureDetection";
+
+type RequestType = (url: string, data: object) => void;
+
+const request: RequestType = (url, data) => {
+  const fn = makeRequester();
+  fn(url, data);
+};
+
+function makeRequester() {
+  if (supportsSendBeacon()) {
+    return requestWithSendBeacon;
+  }
+
+  if (supportsXMLHttpRequest()) {
+    return requestWithXMLHttpRequest;
+  }
+
+  if (supportsNodeHttpModule()) {
+    return requestWithNodeHttpModule;
+  }
+
+  throw new Error(
+    "Could not find a supported HTTP request client in this environment."
+  );
+}
+
+const requestWithSendBeacon: RequestType = (url, data) => {
+  const serializedData = JSON.stringify(data);
+  navigator.sendBeacon(url, serializedData);
+};
+
+const requestWithXMLHttpRequest: RequestType = (url, data) => {
+  const serializedData = JSON.stringify(data);
+  const report = new XMLHttpRequest();
+  report.open("POST", url);
+  report.send(serializedData);
+};
+
+const requestWithNodeHttpModule: RequestType = (url, data) => {
```

Comment:
If we go with two entry point we could link the requester to each build rather than have each requester inside the code. Then we can even imagine exposing an option to provide your own requester at some point.

We'll have to feature detect some of the features still (like `beacon` for example).

Reply:
@samouss Thanks! 
I was struggling to deal with both detecting features and __BROWSER_BUILD__.
Having two entry point makes things much simpler.

Reply:
Now at the entry, `requestFn` is provided to insights class constructor. https://github.com/algolia/search-insights.js/pull/142/commits/2a5cc21db14ab99f33c6bb79746a9a05e5d3bdd7